aten/src/ATen/mps/EmptyTensor.cpp

```diff
@@ -15,38 +16,48 @@ TensorBase empty_mps(
     c10::optional<Device> device_opt,
     c10::optional<bool> pin_memory_opt,
     c10::optional<c10::MemoryFormat> memory_format_opt) {
+#if defined(__APPLE__)
+#if __is_target_os(macOS)
+  if (__builtin_available(macOSApplicationExtension 12.3, *)) {
+    auto device = device_or_default(device_opt);
+    TORCH_INTERNAL_ASSERT_DEBUG_ONLY(device.type() == DeviceType::MPS);
 
-  auto device = device_or_default(device_opt);
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(device.type() == DeviceType::MPS);
+    TORCH_CHECK_NOT_IMPLEMENTED(
+        layout_or_default(layout_opt) == Layout::Strided,
+        "strided tensors not supported yet");
+    check_size_nonnegative(size);
 
-  TORCH_CHECK_NOT_IMPLEMENTED(
-      layout_or_default(layout_opt) == Layout::Strided,
-      "strided meta tensors not supported yet");
-  check_size_nonnegative(size);
+    auto* allocator = at::mps::GetMPSAllocator();
+    int64_t nelements = c10::multiply_integers(size);
+    auto dtype = dtype_or_default(dtype_opt);
+    auto dtype_meta = scalarTypeToTypeMeta(dtype);
+    int64_t size_bytes = nelements * dtype_meta.itemsize();
+    auto storage_impl = c10::make_intrusive<StorageImpl>(
+        c10::StorageImpl::use_byte_size_t(),
+        size_bytes,
+        allocator->allocate(size_bytes),
+        allocator,
+        /*resizeable=*/true);
 
-  auto* allocator = at::mps::GetMPSAllocator();
-  int64_t nelements = c10::multiply_integers(size);
-  auto dtype = dtype_or_default(dtype_opt);
-  auto dtype_meta = scalarTypeToTypeMeta(dtype);
-  int64_t size_bytes = nelements * dtype_meta.itemsize();
-  auto storage_impl = c10::make_intrusive<StorageImpl>(
-      c10::StorageImpl::use_byte_size_t(),
-      size_bytes,
-      allocator->allocate(size_bytes),
-      allocator,
-      /*resizeable=*/true);
+    auto tensor =
+        detail::make_tensor<TensorImpl>(storage_impl, DispatchKey::MPS, dtype_meta);
+    // Default TensorImpl has size [0]
+    if (size.size() != 1 || size[0] != 0) {
+      tensor.unsafeGetTensorImpl()->set_sizes_contiguous(size);
+    }
 
-  auto tensor =
-      detail::make_tensor<TensorImpl>(storage_impl, DispatchKey::MPS, dtype_meta);
-  // Default TensorImpl has size [0]
-  if (size.size() != 1 || size[0] != 0) {
-    tensor.unsafeGetTensorImpl()->set_sizes_contiguous(size);
+    auto memory_format = memory_format_opt.value_or(MemoryFormat::Contiguous);
+    tensor.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
+    return tensor;
+  } else {
+    TORCH_INTERNAL_ASSERT(false, "MPS backend supported MacOS 12.3+")
   }
-
-  auto memory_format = memory_format_opt.value_or(MemoryFormat::Contiguous);
-  tensor.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
-  return tensor;
-
+#else
+  TORCH_INTERNAL_ASSERT(false, "MPS backend supported MacOS 12.3+")
```

Comment:
You can add a `#define MPS_ERROR_NOT_COMPILED` and `#define MPS_ERROR_RUNTIME_TOO_LOW` at the top of this file and re-use them to reduce duplication in error messages.